### PR TITLE
Misc. key-related CLI command UX improvements

### PR DESCRIPTION
**Pull request by spooktheducks**
Numerous key-related UX improvements:
- All key-related commands have been consolidated under the `chainlink keys` subcommand:
    - `chainlink createextrakey`                => `chainlink keys eth create`
    - `chainlink admin info`                    => `chainlink keys eth list`
    - `chainlink node p2p [create|list|delete]` => `chainlink keys p2p [create|list|delete]`
    - `chainlink node ocr [create|list|delete]` => `chainlink keys ocr [create|list|delete]`
    - `chainlink node vrf [create|list|delete]` => `chainlink keys vrf [create|list|delete]`
- Deleting OCR key bundles and P2P key bundles now archives them (i.e., soft delete) so that they can be recovered if needed.  If you want to hard delete a key, pass the new `--hard` flag to the command, e.g. `chainlink keys p2p delete --hard 6`.
- Output from ETH/OCR/P2P/VRF key CLI commands now renders consistently.
- Deleting an OCR/P2P/VRF key now requires confirmation from the user.
- The `--ocrpassword` flag has been removed.  OCR/P2P keys now share the same password at the ETH key (i.e., the password specified with the `--password` flag).
- The OCR and P2P key commands are now remote (API) commands, meaning that they require a running node.

New command output:
```
 λ chainlink keys eth list

🔑 ETH Keys
------------------------------------------------------
Address:    0xff1ae35E9A0584992228d5CF7dC3253D0fA47fF6
ETH:        0.000000000000000000
LINK:       0.000000000000000000
Next nonce: 0
Last used:  
Is funding: false
Created:    2020-10-29 20:04:54.223225 +0000 UTC
Updated:    2020-10-29 20:04:54.223225 +0000 UTC
Deleted:    
------------------------------------------------------
Address:    0xF477A8F490FaD52620D28CA9cfD63145fDcd5E68
ETH:        0.000000000000000000
LINK:       0.000000000000000000
Next nonce: 0
Last used:  
Is funding: true
Created:    2020-10-29 20:04:57.504823 +0000 UTC
Updated:    2020-10-29 20:04:57.504823 +0000 UTC
Deleted:    



 λ chainlink keys p2p list

🔑 P2P Keys
------------------------------------------------------------------------------
ID:         1
Peer ID:    12D3KooWF4APkArgoEgAnBNYjTh5JD56EmT3bfpnABkXpiAUeiiZ
Public key: 0x4dd3403f10dc62e1831308fa66317f7bfbce024c0fc29106e0f9d37eeea9b906
Created:    2020-10-29 20:04:47.221972 +0000 UTC
Updated:    2020-10-29 20:04:47.221972 +0000 UTC
Deleted:    



 λ chainlink keys ocr list

🔑 OCR Keys
-----------------------------------------------------------------------------------------
ID:                    c2d2776ea1c6ff51e179952b7dd16bca49b642688f09f1f358ebdbe1e8826cdb
On-chain signing addr: 0xd402b191f662685f433b14febf427255df4236c2
Off-chain pubkey:      0x21af9d219e66a47879977f0c5eb4b83f3663fdf680c8cee50007ed4b9a462a19
Config pubkey:         e42ceba8e253aef53d3089d4a25b97d08441b002abbba9eb8983219d56e7e177
Created:               2020-10-29 21:37:10.786961 +0000 UTC
Updated:               2020-10-29 21:37:10.786961 +0000 UTC
Deleted:               
-----------------------------------------------------------------------------------------
ID:                    689ee691f66580b48f845f49269269eac709e71be3d9743b3fdb936df8146aa1
On-chain signing addr: 0xaab6a1c3af16d84de9e580509682239d2a6b8a9f
Off-chain pubkey:      0x6af43e1336bc215262974de703d6583f9657d10842e550a626341fad7d5dab51
Config pubkey:         6af128b248937ffa1c949dd384606e2c805679a83eae4f0acde2afb81f89f76b
Created:               2020-10-29 22:02:42.189277 +0000 UTC
Updated:               2020-10-29 22:02:42.189277 +0000 UTC
Deleted:               



 λ chainlink keys vrf list

🔑 VRF Keys
------------------------------------------------------------------------------------------------------------------------------------------------
Compressed:   0x5b859685c745b9875d9b19ef6181bbfa96f447c1da37d153c2c61ba80ce11e4b00
Uncompressed: 0x5b859685c745b9875d9b19ef6181bbfa96f447c1da37d153c2c61ba80ce11e4bba7f1c46ba23f6a43569c82508571a68f9319ec2da779fb20934647e655bb15a
Hash:         0x77eab0fc02d7656f398762fe0807daa557c001eac973f63789deeb38fe3ebe0a
Created:      2020-10-29 21:58:23.244124 +0000 UTC
Updated:      2020-10-29 21:58:23.244124 +0000 UTC
Deleted:      
------------------------------------------------------------------------------------------------------------------------------------------------
Compressed:   0xd1b08daa0af5ff414f067cfa5d9bb786703534daaa706b0f80ff91eb6d396f4500
Uncompressed: 0xd1b08daa0af5ff414f067cfa5d9bb786703534daaa706b0f80ff91eb6d396f45e592e09e637ed2084325cf2b1b5e2cf62660483d7b27190d0971a49959f3df7c
Hash:         0x7ba9d4e41212e66e05120e6eed81d7a88882ad8876b9b4fa664567fac6f420c7
Created:      2020-10-29 21:59:05.555593 +0000 UTC
Updated:      2020-10-29 21:59:05.555593 +0000 UTC
Deleted:      
```

### CODE REVIEW

File: core/cmd/app.go [hunk 1]
```diff
@@ -208,74 +196,30 @@ func NewApp(client *Client) *cli.App {
...
-					Usage:  "Manually set the next nonce for a key. This should NEVER be necessary during normal operation. USE WITH CAUTION: Setting this incorrectly can break your node.",
-					Action: client.SetNextNonce,
-					Flags: []cli.Flag{
-						cli.StringFlag{
-							Name:  "address",
-							Usage: "address of the key for which to set the nonce",
+				cli.Command{
+					Name:   "eth",
+					Usage:  "Local commands for administering the node's Ethereum keys",
+					Hidden: !client.Config.Dev(),
```
Comment (samsondav) [inline core/cmd/app.go:205]: Why is it hidden except in dev mode?

File: core/store/models/key.go [hunk 2]
```diff
@@ -20,8 +20,9 @@ type Key struct {
 	ID        int32 `gorm:"primary_key"`
 	Address   EIP55Address
 	JSON      JSON
-	CreatedAt time.Time `json:"-"`
-	UpdatedAt time.Time `json:"-"`
+	CreatedAt time.Time  `json:"-"`
+	UpdatedAt time.Time  `json:"-"`
+	DeletedAt *time.Time `json:"-"`
```
Comment (samsondav) [inline core/store/models/key.go:25]: Its a pointer here but below you used a `null.Time`?

File: core/store/orm/orm.go [hunk 3]
```diff
@@ -1265,9 +1265,14 @@ func (orm *ORM) KeyExists(address []byte) (bool, error) {
 	return true, err
 }
 
+// ArchiveKey soft-deletes a key whose address matches the supplied bytes.
+func (orm *ORM) ArchiveKey(address []byte) error {
+	return orm.DB.Where("address = ?", address).Delete(models.Key{}).Error
+}
+
 // DeleteKey deletes a key whose address matches the supplied bytes.
 func (orm *ORM) DeleteKey(address []byte) error {
```
Comment (samsondav) [inline core/store/orm/orm.go:1274]: Isn't this method now identical to the one above?

File: core/store/presenters/presenters.go [hunk 4]
```diff
@@ -102,21 +102,27 @@ type balanceable interface {
...
+type ETHKey struct {
 	Address     string       `json:"address"`
 	EthBalance  *assets.Eth  `json:"ethBalance"`
 	LinkBalance *assets.Link `json:"linkBalance"`
+	NextNonce   *int64       `json:"nextNonce"`
+	LastUsed    *time.Time   `json:"lastUsed"`
+	IsFunding   bool         `json:"isFunding"`
+	CreatedAt   time.Time    `json:"createdAt"`
+	UpdatedAt   time.Time    `json:"updatedAt"`
+	DeletedAt   *time.Time   `json:"deletedAt"`
```
Comment (samsondav) [inline core/store/presenters/presenters.go:115]: Again this is a pointer but elsewhere `null.Time` was used.

**Review by samsondav — APPROVED** [review]

File: docs/CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+    - `chainlink createextrakey`                => `chainlink keys eth create`
+    - `chainlink admin info`                    => `chainlink keys eth list`
+    - `chainlink node p2p [create|list|delete]` => `chainlink keys p2p [create|list|delete]`
+    - `chainlink node ocr [create|list|delete]` => `chainlink keys ocr [create|list|delete]`
+    - `chainlink node vrf [create|list|delete]` => `chainlink keys vrf [create|list|delete]`
+- Deleting OCR key bundles and P2P key bundles now archives them (i.e., soft delete) so that they can be recovered if needed.  If you want to hard delete a key, pass the new `--hard` flag to the command, e.g. `chainlink keys p2p delete --hard 6`.
+- Output from ETH/OCR/P2P/VRF key CLI commands now renders consistently.
+- Deleting an OCR/P2P/VRF key now requires confirmation from the user.
+- The `--ocrpassword` flag has been removed.  OCR/P2P keys now share the same password at the ETH key (i.e., the password specified with the `--password` flag).
+
```
Comment (samsondav) [inline docs/CHANGELOG.md:23]: Would it be worth giving them all a consistent ID format?

VRF keys use public key, p2p keys use an integer ID and OCR key bundles use a hash ID 🤦

**Review by samsondav — CHANGES_REQUESTED** [review]

**Review by j16r — DISMISSED** [review]

File: core/store/orm/orm.go (hunk 3 above)
Comment (spooktheducks) [reply core/store/orm/orm.go:1274]: Nah, `DeleteKey` uses `Unscoped`

File: core/store/presenters/presenters.go [hunk 6]
```diff
@@ -102,21 +102,27 @@ type balanceable interface {
...
 }
 
-// AccountBalance holds the hex representation of the address plus it's ETH & LINK balances
-type AccountBalance struct {
+// ETHKey holds the hex representation of the address plus it's ETH & LINK balances
+type ETHKey struct {
 	Address     string       `json:"address"`
 	EthBalance  *assets.Eth  `json:"ethBalance"`
 	LinkBalance *assets.Link `json:"linkBalance"`
+	NextNonce   *int64       `json:"nextNonce"`
```
[the remark this answers is not in the document]
Comment (spooktheducks) [reply core/store/presenters/presenters.go:110]: I'm inclined to agree with you, but I just copied these over verbatim from `models.Key`.  I'm happy to consider changing this but maybe in a separate PR?  Not sure how wide-ranging the ramifications would be.

File: docs/CHANGELOG.md (hunk 5 above)
Comment (spooktheducks) [reply docs/CHANGELOG.md:23]: I think we'd need to have a chat with Lorenz, Peter + node operators, and possibly other folks on Core if we want to change something of this nature

**Review by Bwest981 — APPROVED** [review]

**Pull request merged**